pkg/types/interfacetests/codec_interface_tests.go

```diff
@@ -0,0 +1,377 @@
+package interfacetests
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/smartcontractkit/libocr/commontypes"
+	ocrTypes "github.com/smartcontractkit/libocr/offchainreporting2/types"
+
+	"github.com/smartcontractkit/chainlink-relay/pkg/types"
+)
+
+type EncodeRequest struct {
+	TestStructs  []TestStruct
+	ExtraField   bool
+	MissingField bool
+	TestOn       string
+}
+
+type ChainReaderInterfaceTester interface {
+	Setup(t *testing.T)
+	Teardown(t *testing.T)
+	Name() string
+	EncodeFields(t *testing.T, request *EncodeRequest) ocrTypes.Report
+	GetAccountBytes(i int) []byte
+	GetChainReader(t *testing.T) types.ChainReader
+
+	// IncludeArrayEncodingSizeEnforcement is here in case there's no way to have fixed arrays in the encoded values
+	IncludeArrayEncodingSizeEnforcement() bool
+
+	// SetLatestValue is expected to return the same bound contract and method in the same test
+	// Any setup required for this should be done in Setup.
+	// The contract should take a LatestParams as the params and return the nth TestStruct set
+	SetLatestValue(t *testing.T, testStruct *TestStruct) (types.BoundContract, string)
+}
+
+const (
+	TestItemType       = "TestItem"
+	TestItemSliceType  = "TestItemSliceType"
+	TestItemArray1Type = "TestItemArray1Type"
+	TestItemArray2Type = "TestItemArray2Type"
```

Comment:
Do these need to be exported?

Reply:
all of codec changes have been ripped out

Reply:
coded interface tests have been removed